packages/account/src/providers/transaction-request/helpers.ts

```diff
@@ -0,0 +1,49 @@
+import type { AbstractAddress } from '@fuel-ts/interfaces';
+import { InputType } from '@fuel-ts/transactions';
+
+import type { Resource, ExcludeResourcesOption } from '../resource';
+import { isCoin } from '../resource';
+
+import type {
+  TransactionRequestInput,
+  CoinTransactionRequestInput,
+  MessageTransactionRequestInput,
+} from './input';
+
+export const isRequestInputCoin = (
+  input: TransactionRequestInput
+): input is CoinTransactionRequestInput => input.type === InputType.Coin;
+
+export const isRequestInputMessage = (
+  input: TransactionRequestInput
+): input is MessageTransactionRequestInput => input.type === InputType.Message;
+
+export const isRequestInputResource = (
+  input: TransactionRequestInput
+): input is CoinTransactionRequestInput | MessageTransactionRequestInput =>
+  isRequestInputCoin(input) || isRequestInputMessage(input);
+
```

Comment:
```suggestion
export const isRequestInputResource = (
  input: TransactionRequestInput
): input is CoinTransactionRequestInput | MessageTransactionRequestInput =>
  input.type === InputType.Coin || input.type === InputType.Message;
```

There is no need for the two functions because they aren't used except internally in this file and they're too simple to necessitate an abstraction. This suggestion, if merged as-is, would break `helpers.test.ts`.